src/ApparentHorizons/YlmSpherepack.cpp

```diff
@@ -713,48 +745,49 @@ void YlmSpherepack::interpolate_from_coefs(
 
   const auto& cos_theta = interpolation_info.cos_theta;
 
-  *result = make_with_value<T>(cos_theta, 0.);
-
   // Clenshaw recurrence for m=0.  Separate because there is no phi
   // dependence, and there is a factor of 1/2.
   size_t idx = 0;
   {
-    T ycn = make_with_value<T>(cos_theta, 0.);
-    T ycnp1 = make_with_value<T>(cos_theta, 0.);
+    ycn = make_with_value<T>(cos_theta, 0.);
+    ycnp1 = make_with_value<T>(cos_theta, 0.);
     for (size_t n = n_theta_ - 1; n > 0;
          --n, ++idx) {  // Loops from n_theta_-1 to 1.
-      T ycnp2 = ycnp1;
+      ycnp2 = ycnp1;
       ycnp1 = ycn;
       ycn = cos_theta * alpha[idx] * ycnp1 + beta[idx] * ycnp2 +
             spectral_coefs[a_offset + spectral_stride * index[idx]];
     }
-    *result += 0.5 * interpolation_info.pbar_factor[0] *
-               (beta[idx] * ycnp1 + cos_theta * alpha[idx] * ycn +
-                spectral_coefs[a_offset + spectral_stride * index[idx]]);
+    *result = 0.5 * interpolation_info.pbar_factor[0] *
+              (beta[idx] * ycnp1 + cos_theta * alpha[idx] * ycn +
+               spectral_coefs[a_offset + spectral_stride * index[idx]]);
     ++idx;
   }
   // Now do recurrence for other m.
   for (size_t m = 1; m < l1; ++m) {
-    T ycn = make_with_value<T>(cos_theta, 0.);
-    T ycnp1 = make_with_value<T>(cos_theta, 0.);
-    T ysn = make_with_value<T>(cos_theta, 0.);
-    T ysnp1 = make_with_value<T>(cos_theta, 0.);
+    ycn = make_with_value<T>(num_points, 0.);
```

Comment:
`make_with_value` still does the allocation. You can just do `ycm = 0.` if all you're trying to do is zero it.